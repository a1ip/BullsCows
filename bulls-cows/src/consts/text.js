export const STAGES = {
    guess: 'guess',
    lose: 'lose',
    win: 'win',
    start: 'start'
}

export const FINAL = {
    lose: 'К сожалению, угадать не удалось!',
    win: 'Ура! Код угадан!',
    attempts: 'Итоговое количество попыток: '
}