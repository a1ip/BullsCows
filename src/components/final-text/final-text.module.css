.final {
    padding: 16px;
}

h3 {
    font-size: 1.5rem;
}