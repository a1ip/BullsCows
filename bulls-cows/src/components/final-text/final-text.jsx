import React from 'react'
import style from './final-text.module.css'
import {FINAL} from "../../consts/text"

export const FinalText = ({stage, attempts}) => (
    <div className={style.final}>
        <h3>{FINAL[stage]}</h3>
        <h3>{FINAL.attempts}{attempts}</h3>
    </div>
)