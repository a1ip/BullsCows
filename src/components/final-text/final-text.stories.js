import React from 'react'
import { FinalText } from './final-text'

export default {
    component: FinalText,
    title: 'FinalText',
    parameters: {
        info: {
            text: `
    The Final Screen. Text with information about win or lose and number of attempts.
  `,
        },
    },
}

export const final_text_win = () => <FinalText stage={'win'} attempts={12} />
export const final_text_lose = () => <FinalText stage={'lose'} attempts={5} />