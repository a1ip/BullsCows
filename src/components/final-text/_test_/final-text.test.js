import React from "react"
import renderer from 'react-test-renderer'

import {FinalText} from "../final-text"
import {FINAL} from "../../../consts/text"


describe('Final Text component', () => {

    const FinalTextTest = renderer
        .create(<FinalText stage={'win'} attempts={7}/>)

    it('renders correctly', () => {
        expect(FinalTextTest.toJSON()).toMatchSnapshot()
    })

    it('renders correct text', () => {
        expect(FinalTextTest.root.findAllByType('h3')[0].children).toStrictEqual([FINAL['win']])
        expect(FinalTextTest.root.findAllByType('h3')[1].children).toStrictEqual([FINAL.attempts, "7"])
    })

})