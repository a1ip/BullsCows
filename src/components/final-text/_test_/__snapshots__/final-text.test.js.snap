// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Final Text component renders correctly 1`] = `
<div
  className="final"
>
  <h3>
    Ура! Код угадан!
  </h3>
  <h3>
    Итоговое количество попыток: 
    7
  </h3>
</div>
`;
